### ISSUE TITLE
Hospitality: ThingFilter_Patch running wild when food-restriction window is open

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [OrionFive/Hospitality#539]
**Describe the bug**
When the food-restriction window is open, on a brand-new map, FPS drops from 60 to ~2-3. Dub's Performance Analyzer shows ThingFilter_Patch (wtf? why??) being invoked hundreds of millions of times.

<img width="1947" alt="Screen Shot 2021-05-18 at 18 07 06" src="https://user-images.githubusercontent.com/200/118734614-65122500-b804-11eb-965f-816cec69f595.png">

**To Reproduce**
1. Install and enable only Harmony, Dub's Performance Analyzer, and Rocketman
2. Quickstart a new colony
3. Under restrictions, click "edit" next to a pawn's food-restriction
4. Watch the Harmony Patches graph in DPA

**Console log**
https://gist.github.com/ELLIOTTCABLE/a73345132601a347f6f5d7039d3b0d69

**Comment by OrionFive** [comment]
I'll check it out. I can probably improve this one with a cache.

**Comment by OrionFive** [comment]
I've checked it. There's no point in using a cache, since a cache would be slower than what Hospitality does. This is another case of vanilla not giving a crap and rechecking the whole list of item types every frame. 
I've got a steady 30 fps during this (on a colony), even with the millions of calls, and my computer is 5 years old.
So there might be something else making this worse for you.

But yeah. Not much I can do about it.
